File: 백준/Silver/11723. 집합/집합.py
import sys

input = sys.stdin.readline

M = int(input())
s = set()
lst = ['1','2','3','4','5','6','7','8','9','10','11','12','13','14','15','16','17','18','19','20']
for _ in range(M):
    string = input().rstrip().split()
    if string[0] == 'add':
        s.add(string[1])
    elif string[0] == 'check':
        if string[1] in s:
            print(1)
        else:
            print(0)
    elif string[0] == 'remove':
        if string[1] in s:
            s.remove(string[1])
        else:
            pass
    elif string[0] == 'toggle':
        if string[1] in s:
            s.remove(string[1])
        else:
            s.add(string[1])
    elif string[0] == 'all':
        s = set(lst)
    else:
        s.clear()

File: C++17/백준/Silver/11723. 집합/집합.cc
#include <iostream>
#include <string>
using namespace std;

int main() {
	ios_base::sync_with_stdio(false);
	cin.tie(NULL);
	cout.tie(NULL);
    int m, num = 0;
    cin >> m;

    while (m--) {
        string s;
        int x;
        cin >> s;

        if (s == "add") {
            cin >> x;
            num |= (1 << (x - 1));
        } else if (s == "remove") {
            cin >> x;
            num &= ~(1 << (x - 1));
        } else if (s == "check") {
            cin >> x;
            cout << ((num & (1 << (x - 1))) ? "1\n" : "0\n");
        } else if (s == "toggle") {
            cin >> x;
            num ^= (1 << (x - 1));
        } else if (s == "all") {
            num = (1 << 21) - 1;
        } else num = 0;
    }

    return 0;
}
